home04/app.py

```diff
@@ -0,0 +1,100 @@
+import os
+import threading
+import multiprocessing
+import asyncio
+import time
+import aiofiles
+
+import aiohttp
+import requests
+
+url = ['https://gas-kvas.com/uploads/posts/2023-02/1675495569_gas-kvas-com-p-luchshie-kartinki-dlya-fonovogo-risunka'
+       '-ra-31.jpg',
+       'https://w.forfun.com/fetch/4a/4af0bcc2b0c34fd573eca9f1be9ab245.jpeg',
+       'https://mykaleidoscope.ru/x/uploads/posts/2022-10/1666206241_12-mykaleidoscope-ru-p-kartinka-na-zastavku-oboi'
+       '-12.jpg',
+       'https://mobimg.b-cdn.net/v3/fetch/f4/f4e488ef69ea10573c0ce9cfbaf08643.jpeg',
+       'https://gas-kvas.com/uploads/posts/2023-02/1675446644_gas-kvas-com-p-kartinki-na-fonovii-risunok-rabochego-11'
+       '.jpg']
+
+
+def download(url_img):
+    response = requests.get(url_img).content
+    return response
+
+
+def timed(func):
+    def wrapper():
+        t_start = time.time()
+        func()
+        t_end = time.time()
+        print(f'Method {func.__name__}. Downloaded in {t_end - t_start:.2f} seconds')
+
+    return wrapper
+
+
+def save_image(image: str, type_save_dir: str):
+    image_name = image.split('/')[-1]
+    if not os.path.exists(type_save_dir):
+        os.mkdir(type_save_dir)
+    with open(type_save_dir + '/' + image_name, 'wb') as handler:
```

Comment:
Так правильней. И будет работать не только со строками но и с объектами по классу `Path` из `pathlib` и на любой платформе.
```suggestion
    with open(os.path.join(type_save_dir, image_name), 'wb') as handler:
```